### PR TITLE
[8.x] Don't name unnamed routes inside a group with a prefix

### PR DESCRIPTION
**Pull request by tobyzerner**
Currently if you register an unnamed route within a group with a name prefix, the route's name is set to the prefix.

```php
$router->name('api.')->group(function ($router) {
    $router->get('users', 'UsersController@index');
});

$routes = $router->getRoutes()->get();
echo last($routes)->getName(); // "api."
```

While the consequences of this are somewhat obscure ([but definitely exist](https://github.com/laravel/framework/pull/31917)), it is indeed odd behavior and I would expect `getName()` to return `null` in the above example since I never explicitly set a name on the route.

This PR attempts to fix that within the constraints of the current architecture. I'm not sure if this would be considered a breaking change vs a bug fix, but I can PR this instead to master if desired.

Basically, instead of passing the group's `$action['as']` value into each new route created within it, it passes it as `$action['as_prefix']`. Then when a name is set on a route, it prepends `$action['as_prefix']`.

This also changes the behavior of calling `name` more than once on a route. Previously each time you called it, the value passed would be continually appended:

```php
$router->name('api.')->group(function ($router) {
    $router->get('users', 'UsersController@index')->name('foo')->name('bar') // "api.foobar"
});
```

Now, the value will be appended onto the initial group prefix each time:

```php
$router->name('api.')->group(function ($router) {
    $router->get('users', 'UsersController@index')->name('foo')->name('bar') // "api.bar"
});
```

### CODE REVIEW

**Comment by devcircus** [comment]
Seems like a common sense change but worried it would be a breaking change for a lot of apps. I've see a number of apps with route "groups" for individual routes and using the group name for the route. I know I've ended up with one route in a group after moving other routes out of the group and never refactoring. IMO this should go into the next version with a note in the upgrade guide.

**Comment by driesvints** [comment]
This is a huge breaking change.

**Comment by tobyzerner** [comment]
@taylorotwell would you accept this for master/9.x? Or not interested?
